spark/src/test/scala/org/apache/comet/CometCastSuite.scala

```diff
@@ -55,8 +55,39 @@ class CometCastSuite extends CometTestBase with AdaptiveSparkPlanHelper {
     castTest(generateFloats, DataTypes.IntegerType)
   }
 
-  ignore("cast float to string") {
-    castTest(generateFloats, DataTypes.StringType)
+  test("cast double to string") {
+    val testValues = (
+      Seq(
+        1.0499721536516571e-4,
+        0.001,
+        10000000.0,
+        9999999.0,
+        Float.NegativeInfinity,
+        Float.PositiveInfinity,
+        Float.MinPositiveValue,
+        Double.MinValue,
+        Double.MaxValue,
+        Double.NaN,
+        0.0,
+        -0.0)
+    ).toDF("a")
+    castTest(testValues, DataTypes.StringType, testAnsiModeThrows = false)
```

Comment:
#351 will add some data generation function with random number. May need to rebase on it

Reply:
I will take a look

Reply:
If there are still some inputs that produce different results, we can document that in the compatibility guide and skip fuzz testing (or have some reduced form of fuzz testing). We could also consider adding a config so that the user can opt in or out of enabling this particular cast, depending on the severity of any incompatibilities.

Reply:
Also see https://github.com/apache/datafusion-comet/pull/362 which improves how we handle casts that are not compatible with Spark

Reply:
I rebased #351 will checkout #362